core/chains/evm/txm/orchestrator.go

```diff
@@ -0,0 +1,364 @@
+		if kErr := o.keystore.CheckEnabled(ctx, request.FromAddress, o.chainID); kErr != nil {
+			return tx, fmt.Errorf("cannot send transaction from %s on chain ID %s: %w", request.FromAddress, o.chainID.String(), kErr)
+		}
+
+		var pipelineTaskRunID uuid.NullUUID
+		if request.PipelineTaskRunID != nil {
+			pipelineTaskRunID.UUID = *request.PipelineTaskRunID
+			pipelineTaskRunID.Valid = true
+		}
+
+		if o.fwdMgr != nil && (!utils.IsZero(request.ForwarderAddress)) {
+			fwdPayload, fwdErr := o.fwdMgr.ConvertPayload(request.ToAddress, request.EncodedPayload)
+			if fwdErr == nil {
+				// Handling meta not set at caller.
+				if request.Meta != nil {
+					request.Meta.FwdrDestAddress = &request.ToAddress
+				} else {
+					request.Meta = &txmgrtypes.TxMeta[common.Address, common.Hash]{
+						FwdrDestAddress: &request.ToAddress,
+					}
+				}
+				request.ToAddress = request.ForwarderAddress
+				request.EncodedPayload = fwdPayload
+			} else {
+				o.lggr.Errorf("Failed to use forwarder set upstream: %v", fwdErr.Error())
+			}
+		}
+
+		var meta *sqlutil.JSON
+		if request.Meta != nil {
+			raw, mErr := json.Marshal(request.Meta)
+			if mErr != nil {
+				return tx, mErr
+			}
+			m := sqlutil.JSON(raw)
+			meta = &m
+		}
+
+		wrappedTxRequest := &txmtypes.TxRequest{
+			IdempotencyKey:    request.IdempotencyKey,
+			ChainID:           o.chainID,
+			FromAddress:       request.FromAddress,
+			ToAddress:         request.ToAddress,
+			Value:             &request.Value,
+			Data:              request.EncodedPayload,
+			SpecifiedGasLimit: request.FeeLimit,
+			Meta:              meta,
+			ForwarderAddress:  request.ForwarderAddress,
+
+			PipelineTaskRunID: pipelineTaskRunID,
+			MinConfirmations:  request.MinConfirmations,
+			SignalCallback:    request.SignalCallback,
+		}
+
+		wrappedTx, err = o.txm.CreateTransaction(ctx, wrappedTxRequest)
+		if err != nil {
+			return
+		}
+		o.txm.Trigger(request.FromAddress)
+	}
+
+	if wrappedTx.ID > math.MaxInt64 {
+		return tx, fmt.Errorf("overflow for int64: %d", wrappedTx.ID)
+	}
+
+	tx = txmgrtypes.Tx[*big.Int, common.Address, common.Hash, common.Hash, evmtypes.Nonce, gas.EvmFee]{
+		ID:             int64(wrappedTx.ID),
+		IdempotencyKey: wrappedTx.IdempotencyKey,
+		FromAddress:    wrappedTx.FromAddress,
+		ToAddress:      wrappedTx.ToAddress,
+		EncodedPayload: wrappedTx.Data,
+		Value:          *wrappedTx.Value,
+		FeeLimit:       wrappedTx.SpecifiedGasLimit,
+		CreatedAt:      wrappedTx.CreatedAt,
+		Meta:           wrappedTx.Meta,
+		// Subject: wrappedTx.Subject,
+
+		// TransmitChecker: wrappedTx.TransmitChecker,
+		ChainID: wrappedTx.ChainID,
+
+		PipelineTaskRunID: wrappedTx.PipelineTaskRunID,
+		MinConfirmations:  wrappedTx.MinConfirmations,
+		SignalCallback:    wrappedTx.SignalCallback,
+		CallbackCompleted: wrappedTx.CallbackCompleted,
+	}
+	return
+}
+
+func (o *Orchestrator[BLOCK_HASH, HEAD]) CountTransactionsByState(ctx context.Context, state txmgrtypes.TxState) (uint32, error) {
```

Comment:
Shouldn't this method return the count of transactions in any state? I noticed we only check the unconfirmed map below.

Reply:
Could you maybe add a comment mentioning that here? Just in case someone else comes by this and gets confused